python/pyspark/sql/tests/arrow/test_arrow_udtf.py

```diff
@@ -1410,6 +1410,263 @@ def eval(self, table_arg, struct_arg) -> Iterator["pa.Table"]:
             assert row.table_is_batch is True
             assert row.struct_is_array is True
 
+    def test_arrow_udtf_table_partition_by_single_column(self):
+        @arrow_udtf(returnType="partition_key string, total_value bigint")
+        class PartitionSumUDTF:
+            def eval(self, table_data: "pa.RecordBatch") -> Iterator["pa.Table"]:
+                table = pa.table(table_data)
+
+                # Each partition will have records with the same category
+                if table.num_rows > 0:
+                    category = table.column("category")[0].as_py()
+                    total = pa.compute.sum(table.column("value")).as_py()
+
+                    result_table = pa.table(
+                        {
+                            "partition_key": pa.array([category], type=pa.string()),
+                            "total_value": pa.array([total], type=pa.int64()),
+                        }
+                    )
+                    yield result_table
+
+        self.spark.udtf.register("partition_sum_udtf", PartitionSumUDTF)
+
+        # Create test data with categories
+        test_data = [("A", 10), ("A", 20), ("B", 30), ("B", 40), ("C", 50)]
+        test_df = self.spark.createDataFrame(test_data, "category string, value int")
+        test_df.createOrReplaceTempView("partition_test_data")
+
+        result_df = self.spark.sql(
+            """
+            SELECT * FROM partition_sum_udtf(
+                TABLE(partition_test_data) PARTITION BY category
+            ) ORDER BY partition_key
+        """
+        )
```

Comment:
This is also potentially flaky as same as tests in the previous PR. Use `terminate` to be more stable?